README.md

```diff
@@ -1,51 +1,93 @@
-![astar-cover](https://user-images.githubusercontent.com/40356749/135799652-175e0d24-1255-4c26-87e8-447b192fd4b2.gif)
 
-<div align="center">
+## Add Reward Beneficiary Delegation
 
-[![Integration Action](https://github.com/AstarNetwork/Astar/workflows/Integration/badge.svg)](https://github.com/AstarNetwork/astar-frame/actions)
-[![GitHub tag (latest by date)](https://img.shields.io/github/v/tag/AstarNetwork/Astar)](https://github.com/AstarNetwork/astar-frame/tags)
-[![Substrate version](https://img.shields.io/badge/Substrate-3.0.0-brightgreen?logo=Parity%20Substrate)](https://substrate.dev/)
-[![License](https://img.shields.io/github/license/AstarNetwork/Astar?color=green)](https://github.com/AstarNetwork/astar-frame/LICENSE)
- <br />
-[![Twitter URL](https://img.shields.io/twitter/follow/AstarNetwork?style=social)](https://twitter.com/AstarNetwork)
-[![Twitter URL](https://img.shields.io/twitter/follow/ShidenNetwork?style=social)](https://twitter.com/ShidenNetwork)
-[![YouTube](https://img.shields.io/youtube/channel/subscribers/UC36JgEF6gqatVSK9xlzzrvQ?style=social)](https://www.youtube.com/channel/UC36JgEF6gqatVSK9xlzzrvQ)
-[![Docker](https://img.shields.io/docker/pulls/staketechnologies/astar-collator?logo=docker)](https://hub.docker.com/r/staketechnologies/astar-collator)
-[![Discord](https://img.shields.io/badge/Discord-gray?logo=discord)](https://discord.gg/Z3nC9U4)
-[![Telegram](https://img.shields.io/badge/Telegram-gray?logo=telegram)](https://t.me/PlasmOfficial)
-[![Medium](https://img.shields.io/badge/Medium-gray?logo=medium)](https://medium.com/astar-network)
+## Overview
+To send the accumulated rewards to other accounts, stakers can now change the reward destination to DelegateBalance using `set_reward_destination` and then add delegatee_address using `set_delegatee` extrinsic. This must be done for all contract_id for which staker expect to receive rewards.
 
-</div>
+**WARNING: Once a staker changes RewardDestination from DelegateBalance to other, its delegatee_account information is deleted for all contract_ids and must be set up again if switched back to DelegateBalance.**
 
-Astar Network is an interoperable blockchain based the Substrate framework and the hub for dApps within the Polkadot Ecosystem.
-With Astar Network and Shiden Network, people can stake their tokens to a Smart Contract for rewarding projects that provide value to the network.
+## Modified Data Structure
+In order to delegate collected rewards to Beneficiary in DappStaking, we introduced **DelegateBalance** variant in `RewardDestination` 
+```
+pub enum RewardDestination {
+    /// Rewards are transferred to stakers free balance without any further action.
+    FreeBalance,
+    /// Rewards are transferred to stakers balance and are immediately re-staked
+    /// on the contract from which the reward was received.
+    StakeBalance,
+    /// Rewards are transfered to delegatee account's free balance
+    DelegateBalance,
```

Comment:
Nice approach!

A trick question - would this require storage migration for live chains? And why? :slightly_smiling_face:

Reply:
No this would not require storage migration in my opinion.
The only storage item that is affected by the change is :
```
pub type Ledger<T: Config> =
        StorageMap<_, Blake2_128Concat, T::AccountId, AccountLedger<BalanceOf<T>>, ValueQuery>;
```
We have just added a variant to Enum(RewardDestination) and no change to existing state is introduced.